### PR TITLE
V dmkap/serenity/guest

### PR DESCRIPTION
**Pull request by v-dmkapi**
Please make sure every Pull Request has the following information:

<ul>
  <li>Guest scenario with prompt auto</li>
  <li>Adding cloud switch and including logic for wrong API return.</li>
  <li>@mentions of the person required for reviewing proposed changes (optional)</li>
</ul>

### CODE REVIEW

File: automation/src/test/java/com/microsoft/identity/common/test/automation/AcquireTokenPromptAutoGuestTest.java [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+
+    @Test
+    public void should_be_able_to_acquire_token() {
+
+        james.attemptsTo(
+                acquireToken.withPrompt("Auto"),
+                clickDone,
+                readCache);
+
+        then(james).should(seeThat(TokenCacheItemCount.displayed(), is(6)));
```
Comment (kreedula) [inline automation/src/test/java/com/microsoft/identity/common/test/automation/AcquireTokenPromptAutoGuestTest.java]: This wouldn't work with latest common cache changes, the token cache count is 8 for common and 6 for Adal legacy. My PR here takes care of that  https://github.com/AzureAD/azure-activedirectory-library-for-android/pull/1213/.
Once merged you can reference that and make the change

File: automation/src/test/java/com/microsoft/identity/common/test/automation/utility/TestConfigurationHelper.java [hunk 2]
```diff
@@ -35,6 +35,11 @@
 
     public static TestConfiguration GetTestConfiguration(TestConfigurationQuery query) {
 
+        //Cloud set
+        if(query.federationProvider == "Cloud") {
+            query.federationProvider = " ";
```
Comment (kreedula) [inline automation/src/test/java/com/microsoft/identity/common/test/automation/utility/TestConfigurationHelper.java]: Why is this needed ?

Comment (v-dmkapi) [reply automation/src/test/java/com/microsoft/identity/common/test/automation/utility/TestConfigurationHelper.java]: Your right this wouldn't be needed for this test case. I'll remove until its needed.
